src/tests/constructors.test.ts

```diff
@@ -240,8 +242,8 @@ describe('withSchema', () => {
   })
 
   it('uses zod parsers to parse the input and context and call the schema function', async () => {
-    const parser = z.object({ id: z.preprocess(Number, z.number()) })
-    const ctxParser = z.object({ uid: z.preprocess(Number, z.number()) })
+    const parser = z.object({ id: z.coerce.number() })
+    const ctxParser = z.object({ uid: z.coerce.number() })
```

Comment:
For some reason the standard-schema was not working with the preprocess.
We can discuss/investigate it further.

Reply:
Besides this issue I don't see any breaking changes, am I missing something?

Reply:
@diogob the "breaking change" is that standard schema only works with zod@3.24+

Reply:
Breaking changes:
- Zod at 3.24+
- preprocess seems to be broken
- We don't export `ParserSchema` anymore